nomenclature/__init__.py

```diff
@@ -265,3 +282,14 @@ def _validate_directional(x):
     """Utility function to check whether region-to-region code is valid"""
     x = x.split('>')
     return len(x) == 2 and all([i in regions for i in x])
+
+
+def _validate_unit(x):
+    # sub function to filter out variables with valid name
+    for i in reversed(x):  # iterate list reversely due to 'remove' method
+        if i in variables.keys():
+            logger.warning(
+                'Unit for variable %s is not given in %s.', i, variables[
+                    i]['unit'])
```

Comment:
This is deprecated way to format messages, f-strings are easier to read

```suggestion
                f'Unit for variable {i} is not given in {variables[i]['unit']}.'
```